refactor(ListItem): Moved ListItem into its own dir

ElliotTheHuman opened the pull request:
## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [ ] ♿️ Accessibility addressed
- [ ] 🌏 Localization & Internationalization addressed
- [ ] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 🧳 Bundle size impact addressed
- [ ] 🏁 Performance impacts addressed
- [ ] 👾 Browsers tested
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

lukelooker commented on packages/components/src/ListItem/utils/getDetailOptions.ts:
```diff
@@ -24,7 +24,7 @@
  */
 
-import { Detail } from '../types'
+import { Detail } from '../../ListItem/types'
```
```suggestion
import { Detail } from '../types'
```

lukelooker commented on packages/components/src/ListItem/utils/listItemBackgroundColor.ts:
```diff
@@ -26,7 +26,7 @@
 import { itemSelectedColor, Theme } from '@looker/design-tokens'
 import { css } from 'styled-components'
-import { FlexibleColor, ListItemStatefulProps } from '../types'
+import { FlexibleColor, ListItemStatefulProps } from '../../ListItem/types'
```
```suggestion
import { FlexibleColor, ListItemStatefulProps } from '../types'
```

lukelooker commented on packages/components/src/ListItem/utils/listItemColor.ts:
```diff
@@ -28,7 +28,7 @@ import {
   listItemColorAppliesToLabel,
   listItemColorOptions,
-} from '../types'
+} from '../../ListItem/types'
```
```suggestion
} from '../types'
```

lukelooker commented on packages/components/src/ListItem/utils/listItemDimensions.ts:
```diff
@@ -24,7 +24,7 @@
  */
 
-import { DensityRamp, ListItemDimensions } from '../types'
+import { DensityRamp, ListItemDimensions } from '../../ListItem/types'
```
```suggestion
import { DensityRamp, ListItemDimensions } from '../types'
```

lukelooker commented on packages/components/src/ListItem/utils/listItemPadding.ts:
```diff
@@ -26,7 +26,7 @@
 import { SpaceRamp } from '@looker/design-tokens'
 import { css } from 'styled-components'
-import { DensityRamp } from '../types'
+import { DensityRamp } from '../../ListItem/types'
```
```suggestion
import { DensityRamp } from '../types'
```

lukelooker commented on packages/components/src/Menu/MenuList.story.tsx:
```diff
@@ -30,7 +30,7 @@ import { Create } from '@styled-icons/material-outlined/Create'
 import { Undo } from '@styled-icons/material-outlined/Undo'
 import { Box, Grid, Space } from '../Layout'
-import { DensityRamp } from '../List/types'
+import { DensityRamp } from '../ListItem/types'
```
Not as part of this PR but we should log a ticket to pull `DensityRamp` up to @looker/design-tokens soon.

lukelooker commented on packages/components/src/Tree/TreeCollectionContext.ts:
```diff
@@ -25,7 +25,7 @@
 
 import { createContext } from 'react'
-import { DensityProps } from '../List/types'
+import { DensityProps } from '../ListItem/types'
```
These should also find their way to @looker/design-tokens

lukelooker commented on packages/components/src/Tree/TreeContext.tsx:
```diff
@@ -25,7 +25,7 @@
 
 import { createContext } from 'react'
-import { DensityProps, ListColor } from '../List/types'
+import { DensityProps, ListColor } from '../ListItem/types'
```
Should `ListColor` be in `/List/types`?

ElliotTheHuman replied:
🤦  Yes, that would make a lot of sense.

ElliotTheHuman replied:
I opted to just put `ListColor` in `/List/List.tsx` since I _think_ it's the only type we need to move from `/ListItem/type.ts`.

lukelooker commented on packages/components/src/Tree/TreeStyle.tsx:
```diff
@@ -29,15 +29,15 @@ import styled, { css } from 'styled-components'
 import { Accordion2Content } from '../Accordion2/Accordion2Content'
-import { FlexibleColor, ListItemStatefulProps } from '../List/types'
-import { listItemBackgroundColor } from '../List/utils'
+import { FlexibleColor, ListItemStatefulProps } from '../ListItem/types'
```
Perhaps we should export the types from the `LisItemt/index.ts`?

ElliotTheHuman commented:
- Moved `ListColor` and `ListColorProps` over to `/List/types.ts`.
- Exported all of `./ListItem/types.ts` from `./ListItem/index.ts`
  - This caused a name conflict with `Select's DetailOptions` so I renamed `ListItem's DetailOptions` to `ListItemDetailOptions`

lukelooker reviewed (approved):
LGTM!
